src/muistot/backend/repos/project.py

```diff
@@ -183,62 +201,65 @@ async def construct_project(self, m) -> Project:
                     LEFT JOIN users au JOIN project_admins pa ON pa.user_id = au.id ON au.username = :user
                     """,
                     """
                      AND p.published OR au.id IS NOT NULL OR su.id IS NOT NULL
-                    """
+                    """,
                 )
-                if self.authenticated else
-                self._select % ("", "", " AND p.published"),
-                values=
-                dict(lang=self.lang)
-                if not self.authenticated else
-                dict(lang=self.lang, user=self.identity)
-
+                if self.authenticated
+                else self._select % ("", "", " AND p.published"),
+                values=dict(lang=self.lang)
+                if not self.authenticated
+                else dict(lang=self.lang, user=self.identity),
             )
-            if m is not None and (
-                    ProjectRepo._check_dates(m)
-                    or (self.superuser or m["is_admin"] if self.authenticated else False)
+            if m is not None
+            and (
+                ProjectRepo._check_dates(m)
+                or (self.superuser or m["is_admin"] if self.authenticated else False)
             )
         ]
 
-    @append_identifier('project', value=True)
+    @append_identifier("project", value=True)
     @require_status(Status.PUBLISHED, Status.EXISTS | Status.ADMIN)
     async def one(self, project: PID, status: Status = None) -> Project:
         m = await self.db.fetch_one(
             self._select % ("", "", " AND p.name = :project"),
-            values=dict(lang=self.lang, project=project)
+            values=dict(lang=self.lang, project=project),
         )
         if m is None:
             raise HTTPException(
                 status_code=HTTP_406_NOT_ACCEPTABLE,
-                detail='Project missing default localization'
+                detail="Project missing default localization",
             )
         elif not ProjectRepo._check_dates(m) and Status.ADMIN not in status:
             raise HTTPException(
-                status_code=HTTP_404_NOT_FOUND,
-                detail='Project not found'
+                status_code=HTTP_404_NOT_FOUND, detail="Project not found"
             )
         out = await self.construct_project(m)
         return out
 
-    @append_identifier('project', key='id')
-    @require_status(Status.DOES_NOT_EXIST | Status.SUPERUSER, errors={
-        Status.DOES_NOT_EXIST | Status.AUTHENTICATED: HTTPException(
-            status_code=HTTP_403_FORBIDDEN,
-            detail="Not enough privileges",
-        )
-    })
+    @append_identifier("project", key="id")
+    @require_status(
+        Status.DOES_NOT_EXIST | Status.AUTHENTICATED,
+        errors={
+            Status.DOES_NOT_EXIST
+            | Status.AUTHENTICATED: HTTPException(
+                status_code=HTTP_403_FORBIDDEN,
+                detail="Not enough privileges",
+            )
+        },
+    )
```

Comment:
The error statuses are the same as the acceptable statuses here, so in my eyes the error statuses should either be removed or changed to `Status.DOES_NOT_EXIST | Status.ANONYMOUS`.

Reply:
It's true, I'll make it happen